internal/service/advancedclustertpf/resource.go

```diff
@@ -55,40 +68,24 @@ func (r *rs) Schema(ctx context.Context, req resource.SchemaRequest, resp *resou
 func (r *rs) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
 	var plan TFModel
 	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
-	if resp.Diagnostics.HasError() {
-		return
-	}
-	sdkReq := NewAtlasReq(ctx, &plan, &resp.Diagnostics)
-	if resp.Diagnostics.HasError() {
-		return
-	}
-	err := StoreCreatePayload(sdkReq)
-	if err != nil {
-		resp.Diagnostics.AddError("errorCreate", fmt.Sprintf(errorCreate, err.Error()))
+	diags := &resp.Diagnostics
+	if diags.HasError() {
 		return
 	}
-	tfNewModel, shouldReturn := mockedSDK(ctx, &resp.Diagnostics, plan.Timeouts)
-	if shouldReturn {
-		return
+	summary, detail := r.createCluster(ctx, &plan, diags, resp)
+	if summary != "" || detail != "" {
+		diags.AddError(summary, detail)
 	}
-	resp.Diagnostics.Append(resp.State.Set(ctx, tfNewModel)...)
 }
 
 func (r *rs) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
 	var state TFModel
-	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
-	if resp.Diagnostics.HasError() {
+	diags := &resp.Diagnostics
+	diags.Append(req.State.Get(ctx, &state)...)
+	if diags.HasError() {
 		return
 	}
-	if state.ClusterID.IsNull() {
-		tfModel, shouldReturn := mockedSDK(ctx, &resp.Diagnostics, state.Timeouts)
-		if shouldReturn {
-			return
-		}
-		resp.Diagnostics.Append(resp.State.Set(ctx, tfModel)...)
-	} else {
-		resp.Diagnostics.Append(resp.State.Set(ctx, state)...)
-	}
+	r.readCluster(ctx, &state, &resp.State, diags)
 }
 
 func (r *rs) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
```

Comment:
I love how Create and Read are, few lines and high level, delegating in the function doing the hard work.   We might want to use a similar approach for Update

Reply:
Update changed in 426f1b19bbeb918db24d8f0249ddf05547d6f97e
WDYT?

Reply:
i like it better, thanks